pkg/controller/migplan/validation.go

```diff
@@ -474,6 +483,36 @@ func (r ReconcileMigPlan) validateDestinationCluster(plan *migapi.MigPlan) error
 	return nil
 }
 
+func (r ReconcileMigPlan) validateOperatorVersions(plan *migapi.MigPlan) error {
+	destRef := plan.Spec.DestMigClusterRef
+	srcRef := plan.Spec.SrcMigClusterRef
+
+	srcCluster, err := migapi.GetCluster(r, srcRef)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+
+	destCluster, err := migapi.GetCluster(r, destRef)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+	srcClusterCondition := srcCluster.Status.HasCondition(migcluster.OperatorVersionMismatch)
+	destClusterCondition := destCluster.Status.HasCondition(migcluster.OperatorVersionMismatch)
+
+	if !srcClusterCondition && !destClusterCondition {
```

Comment:
Shouldn't the logic be flipped here? Seems like this is issuing a warning when neither cluster is mismatched.